### PR TITLE
MIC-2307: Add single worker log for all performance log messages

### PR DESCRIPTION
**Pull request by mattkappel**
This change adds a single worker log for all workers to write
the JSON-formatted performance summary log messages to, for
consumption by vipin.

Tested against a real workload (LSFF) with performance.log and
vipin output looking as expected.

### CODE REVIEW

File: src/vivarium_cluster_tools/vipin/log_parser.py [hunk 1]
```diff
@@ -10,82 +9,68 @@
...
-def parse_log_directory(input_directory: Union[Path, str], output_directory: Union[Path, str], output_hdf: bool,
-                        random_draw: int = 0):
+def parse_log_directory(input_directory: Union[Path, str], output_directory: Union[Path, str], output_hdf: bool):
     input_directory, output_directory = Path(input_directory), Path(output_directory)
-    worker_log_pattern = re.compile(r'^([0-9]+)\.([0-9]+)\.log$')
-    log_files = [f for f in input_directory.iterdir() if worker_log_pattern.fullmatch(f.name)]
-
-    if random_draw:
-        log_files = sample(log_files, random_draw)
+    perf_log = PerformanceLog(Path(input_directory / 'performance.log'))
```
Comment (collijk) [inline src/vivarium_cluster_tools/vipin/log_parser.py:34]: `input_directory / 'performance.log'` is a `Path`.  No need to coerce.

File: src/vivarium_cluster_tools/vipin/log_parser.py [hunk 2]
```diff
@@ -10,82 +9,68 @@
...
 
-    worker_data = []
-    for i, file in enumerate(log_files):
-        # TODO: set up tdqm progress bar
-        logger.info(f'Parsing file {i+1} of {len(log_files)}.')
-        log = WorkerLog(file)
-        for item in log.get_summaries():
-            worker_data.append(item)
-    worker_df = pd.concat(worker_data)
+    perf_data = []
```
Comment (collijk) [inline src/vivarium_cluster_tools/vipin/log_parser.py:36]: Why not a `perf_log.to_df()` method?

File: src/vivarium_cluster_tools/vipin/log_parser.py (hunk 1 above)
Comment (mattkappel) [reply src/vivarium_cluster_tools/vipin/log_parser.py:34]: Oh good point.

File: src/vivarium_cluster_tools/vipin/log_parser.py (hunk 2 above)
Comment (mattkappel) [reply src/vivarium_cluster_tools/vipin/log_parser.py:36]: That's a good suggestion. I'll change that.

File: src/vivarium_cluster_tools/psimulate/distributed_worker.py [hunk 3]
```diff
@@ -225,3 +208,26 @@ def worker(parameters: Mapping):
         raise
     finally:
         logger.info('Exiting job: {}'.format((input_draw, random_seed, model_specification_file, branch_config)))
+
+
+def do_sim_epilogue(end_snapshot, event, exec_time, parameters, start_snapshot):
```
Comment (collijk) [inline src/vivarium_cluster_tools/psimulate/distributed_worker.py:213]: Typing on the args here would be super helpful.

File: src/vivarium_cluster_tools/vipin/log_parser.py [hunk 4]
```diff
@@ -10,82 +9,68 @@
...
-        name = file.stem.split('.')  # TODO: validate name is <sge job id>.<task id>.log
-        self.sge_job_id = int(name[0])
-        self.task_id = int(name[1])
+        self.count = 0
 
     def get_summaries(self) -> dict:
-        """Get all performance summary log messages in WorkerLog"""
-        telemetry_pattern = re.compile(r'^\{\"host\".+\"job_number\".+\}$')
+        """Get all performance summary log messages in PerformanceLog"""
+        telemetry_pattern = re.compile(r'^{\"host\".+\"job_number\".+}$')
```
Comment (collijk) [inline src/vivarium_cluster_tools/vipin/log_parser.py:19]: You're telemetry pattern is probably better as a class constant.

File: src/vivarium_cluster_tools/vipin/log_parser.py [hunk 5]
```diff
@@ -10,82 +9,68 @@
...
 
-        # Ideally we'd only need to look at the tail of the file, but because a worker can do multiple draws, we need
-        # to iterate over all the lines...
-        # TODO: either change worker to cut a new log for each draw and check tail reversed(list(log))
-        #  *OR* parallelize this
+        self.count = 0
```
Comment (collijk) [inline src/vivarium_cluster_tools/vipin/log_parser.py:21]: Count, on the other hand, looks like it should be a local variable.

Comment (mattkappel) [reply src/vivarium_cluster_tools/vipin/log_parser.py:21]: There isn't strong purpose for the count, it's unused. My original intent was to report the number of entries, but the count is redundant to, say, df.shape. I'll remove it.

File: src/vivarium_cluster_tools/vipin/log_parser.py [hunk 6]
```diff
@@ -10,82 +9,68 @@
...
         f = self.file.open('r')
```
Comment (collijk) [inline src/vivarium_cluster_tools/vipin/log_parser.py:22]: You always want to use a context manager pattern for file opening (and pretty much every other situation that looks like 1. acquire resource, 2. use resource, 3. free resource).  

```
with self.file.open() as f:  # Will default open for read
   ... do stuff with f ...
# Exiting the block guarantees things are cleaned up
```

**Review by collijk — APPROVED** [review]
Overall looks good!

**Pull request merged**